src/menu/custom_menu_system.c

```diff
@@ -96,7 +96,11 @@ void custom_menu_system_init(void) {
 }
 
 void custom_menu_destroy(void) {
-    /* TODO: we should probably clean up all of this stuff */
+    free(sHead);
```

Comment:
>      free(sHead);
This doesn't quite clean up all of the `calloc()`'d memory. You would have to free `sHead->me` as well as recursively call `free()` on all of the child buttons/menu/labels and their buttons/menus/labels. I think we're better off with just replacing this line with a comment for now:
`TODO: clean up all of the calloc()'d memory
It's very tricky to get right, and it's not enough memory to worry about. It might not be worth cleaning up at all in case we ever allow a player to exit their connected game and return to the main menu

Reply:
The buttons are out of scope so ya there would be a lot more to do here.